discovery_engine_core/web-api/openapi.yaml

```
@@ -128,19 +128,206 @@ paths:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DocumentPropertyResponse'
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id or property id not found
+    put:
+      tags:
+        - documents
+      summary: Set a document property
+      description: Sets or replaces the property of the document.
+      parameters:
+        - name: document_id
+          in: path
+          description: Id of the document
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentId'
+        - name: property_id
+          in: path
+          description: Id of the document property
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentPropertyId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DocumentPropertyRequest'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id not found
+    delete:
+      tags:
+        - documents
+      summary: Delete a document property
+      description: Deletes the property of the document.
+      parameters:
+        - name: document_id
+          in: path
+          description: Id of the document
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentId'
+        - name: property_id
+          in: path
+          description: Id of the document property
+          required: true
+          schema:
+            $ref: '#/components/schemas/DocumentPropertyId'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid document id or property id
+        '404':
+          description: document id or property id not found
+
 components:
   schemas:
     DocumentId:
       description: A document id can be any non-empty UTF-8 string that does not contain the null byte.
       type: string
       example: "document id"
+    DocumentPropertyId:
+      description: A document property id can be any non-empty UTF-8 string that does not contain the null byte.
+      type: string
+      example: "document property id"
+    DocumentProperty:
+      description: Arbitrary property that can be attached to a document.
+      example: "News title"
+    DocumentPropertyRequest:
+      type: object
+      required: [property]
+      properties:
+        property:
+          $ref: '#/components/schemas/DocumentProperty'
+    DocumentPropertyResponse:
+      type: object
+      required: [property]
+      properties:
+        property:
+          $ref: '#/components/schemas/DocumentProperty'
     DocumentProperties:
       description: Arbitrary properties that can be attached to a document.
```

:art:
Since we cannot specify this with the type we add it to the description
```suggestion
      description: Arbitrary properties that can be attached to a document. A key must be a valid `DocumentPropertyId`.
```